controlplane/kubeadm/webhooks/scale.go

```diff
@@ -0,0 +1,80 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package webhooks
+
+import (
+	"context"
+	"net/http"
+
+	autoscalingv1 "k8s.io/api/autoscaling/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/cluster-api/controlplane/kubeadm/api/v1beta1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+)
+
+// +kubebuilder:webhook:verbs=create;update,path=/validate-scale-controlplane-cluster-x-k8s-io-v1beta1-kubeadmcontrolplane,mutating=false,failurePolicy=fail,matchPolicy=Equivalent,groups=controlplane.cluster.x-k8s.io,resources=kubeadmcontrolplanes/scale,versions=v1beta1,name=validation-scale.kubeadmcontrolplane.controlplane.cluster.x-k8s.io,sideEffects=None,admissionReviewVersions=v1;v1beta1
+
+// KCPScaleValidator validates KCP for replicas.
+type KCPScaleValidator struct {
+	Client  client.Client
+	decoder *admission.Decoder
+}
+
+// Handle will validate for number of replicas.
+func (v *KCPScaleValidator) Handle(ctx context.Context, req admission.Request) admission.Response {
+	kcpScale := &autoscalingv1.Scale{}
+
+	err := v.decoder.Decode(req, kcpScale)
+	if err != nil {
+		return admission.Errored(http.StatusBadRequest, err)
+	}
+
+	kcp := &v1beta1.KubeadmControlPlane{}
+	kcpKey := types.NamespacedName{Namespace: kcpScale.ObjectMeta.Namespace, Name: kcpScale.ObjectMeta.Name}
+	if err = v.Client.Get(ctx, kcpKey, kcp); err != nil {
+		return admission.Errored(http.StatusInternalServerError, err)
+	}
+
+	if kcpScale.Spec.Replicas == 0 {
+		return admission.Denied("replicas cannot be zero")
+	}
```

Comment:
```suggestion
	if kcpScale.Spec.Replicas <= 0 {
		return admission.Denied("replicas cannot be less than or equal to 0")
	}
```
I think that would be better, to also cover the <0 case

Reply:
I found that `kubectl scale` does not allow --replicas < 0. Do you think, It's still better to handle this validation?
```
k scale kubeadmcontrolplane capi-quickstart-control-plane --replicas=-1    
error: The --replicas=COUNT flag is required, and COUNT must be greater than or equal to 0`
```

Reply:
I'm not sure if this validation is client or server-side, so I thought let's validate to be on the safe side

Reply:
But yeah they probably check it on the server-side. I'm fine with just checking for 0. But let's use `0` instead of `zero` in the error message.